engine/src/test/java/org/hibernate/validator/test/cfg/ProgrammaticConstraintDefinitionsTest.java

```diff
@@ -66,6 +68,50 @@ public void safeHtmlProgrammaticDefinition() {
 		doProgrammaticTest( new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.RELAXED ).baseURI( "http://localhost" ),
 				"<img src='/some/relative/url/image.png' />", 0
 		);
+
+		doProgrammaticTest(
+				new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE )
+						.additionalTag( "img" ).attribute( "src" ),
+				"<img src='data:image/png;base64,100101' />", 1
+		);
+		doProgrammaticTest(
+				new SafeHtmlDef().whitelistType( SafeHtml.WhiteListType.NONE )
+						.additionalTag( "img" ).attribute( "src" ).protocols( "data" ),
+				"<img src='data:image/png;base64,100101' />", 0
+		);
+		doProgrammaticTest(
```

Comment:
@marko-bekhta @gsmet Can you take a look at this PR: https://github.com/hibernate/hibernate-validator/pull/739. It's based on your's but takes a slightly different approach for handling the inner types of `@SafeHtml` in the API. Instead of having one fluent invocation on one level this adds `new TagDef()` and new `AttributeDef()` for those. For the common case of no further configuration on the next level, an overloaded version of just taking a string is added.

It's failing the test, i.e. I must have done something wrong within the implementation, but before going further I wanted to share it with you to discuss on the API. I like that it makes clearer, when one element has been fully configured as opposed to one huge fluent invocation. WDYT?